### PR TITLE
Port #51572 to master

### PR DESCRIPTION
**Pull request by terminalmage**
See https://github.com/saltstack/salt/pull/51572

### CODE REVIEW

File: salt/modules/boto_route53.py [hunk 1]
```diff
@@ -106,6 +106,10 @@ def _get_split_zone(zone, _conn, private_zone):
     return False
 
 
+def _is_retryable_error(exception):
+    return exception.code not in ["SignatureDoesNotMatch"]
```
Comment (DmitryKuzmenko) [inline salt/modules/boto_route53.py:110]: Just a question: is there a reason to look up a single-item list? Maybe just use `!=`?

Comment (terminalmage) [reply salt/modules/boto_route53.py:110]: Good question. The original code author is @kaidokert, I just ported this to master.

Comment (terminalmage) [reply salt/modules/boto_route53.py:110]: I have changed this to `!=` and am working on unit tests.

**Comment by terminalmage** [comment]
Tests added

**Review by dwoz — APPROVED** [review]

**Pull request merged**